model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -170,7 +170,7 @@ class ModelReplicationServer(
             fun ApplicationCall.repositoryId() = RepositoryId(parameters["repository"]!!)
             fun PipelineContext<Unit, ApplicationCall>.repositoryId() = call.repositoryId()
 
-            val useRoleIds = call.request.queryParameters["useRoleIds"] != "false"
+            val useRoleIds = call.request.queryParameters["useRoleIds"]?.toBooleanStrict() != false
```

Comment:
`toBooleanStrict` throws an `IllegalArgumentException`. This results in HTTP status 500 (Internal Server Error) and not (400 Bad Request).

Reply:
Fixed the broken behavior in [f500745](https://github.com/modelix/modelix.core/pull/692/commits/f50074594b160985ca753a2f43f125623bce9ccc)

Reply:
`call.request.queryParameters["useRoleIds"]?.toBooleanStrictOrNull() != false` has exatly the same result as `val useRoleIds = call.request.queryParameters["useRoleIds"] != "false"` just with more overhead...

```java
@SinceKotlin("1.5")
public fun String.toBooleanStrictOrNull(): Boolean? = when (this) {
    "true" -> true
    "false" -> false
    else -> null
}
```

---

If you don't want to make the parsing stricter then let us only update the `model-server.yaml`.